scopesim/optics/optical_train.py

```diff
@@ -292,22 +298,164 @@ def readout(self, filename=None, **kwargs):
 
         """
 
-        hdus = []
+        hduls = []
         for i, detector_array in enumerate(self.detector_arrays):
             array_effects = self.optics_manager.detector_array_effects
             dtcr_effects = self.optics_manager.detector_effects
-            hdu = detector_array.readout(self.image_planes, array_effects,
+            hdul = detector_array.readout(self.image_planes, array_effects,
                                          dtcr_effects, **kwargs)
 
+            try:
+                hdul = self.write_header(hdul)
+            except Exception as error:
+                print("\nWarning: header update failed, data will be saved with incomplete header.")
+                print("Reason: ", sys.exc_info()[0], error)
+                print("")
+
             if filename is not None and isinstance(filename, str):
                 fname = filename
                 if len(self.detector_arrays) > 1:
                     fname = str(i) + "_" + filename
-                hdu.writeto(fname, overwrite=True)
-
-            hdus += [hdu]
-
-        return hdus
+                hdul.writeto(fname, overwrite=True)
+
+            hduls += [hdul]
+
+        return hduls
+
+    def write_header(self, hdulist):
+        """Writes meaningful header to simulation product"""
+        hdulist[3].header
```

Comment:
why is this here?

Reply:
Oh. This was to test the try/except wrapper by raising an error and I forgot to remove it before committing (it was late...). I'll remove it instantly.